spanner/database_leader_placement_samples.rb

```diff
@@ -0,0 +1,250 @@
+                                          database: database_id
+  statements = [
+    "ALTER DATABASE `#{database_id}` SET OPTIONS (
+      default_leader = '#{default_leader}'
+    )"
+  ]
+
+  job = db_admin_client.update_database_ddl database: db_path,
+                                            statements: statements
+
+  job.wait_until_done!
+  database = job.results
+
+  puts "Updated default leader"
+  # [END spanner_update_database_with_default_leader]
+end
+
+def database_ddl project_id:, instance_id:, database_id:
+  # [START spanner_get_database_ddl]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID"
+
+  require "google/cloud/spanner"
+
+  db_admin_client = \
+    Google::Cloud::Spanner::Admin::Database::V1::DatabaseAdmin::Client.new
+
+  db_path = db_admin_client.database_path project: project_id,
+                                          instance: instance_id,
+                                          database: database_id
+  ddl = db_admin_client.get_database_ddl database: db_path
+
+  puts ddl.statements
+
+  # [END spanner_get_database_ddl]
+end
+
+def query_information_schema_database_options \
+  project_id:, instance_id:, database_id:
+  # [START spanner_query_information_schema_database_options]
+  # project_id  = "Your Google Cloud project ID"
+  # instance_id = "Your Spanner instance ID"
+  # database_id = "Your Spanner database ID
+
+  require "google/cloud/spanner"
+
+  spanner = Google::Cloud::Spanner.new project: project_id
+  client = spanner.client instance_id, database_id
+
+  client.execute(
+    "SELECT s.OPTION_NAME, s.OPTION_VALUE " \
+    "FROM INFORMATION_SCHEMA.DATABASE_OPTIONS s" \
+    " WHERE s.OPTION_NAME = 'default_leader'"
+  ).rows.each do |row|
+    puts row
+  end
+
+  # [END spanner_query_information_schema_database_options]
+end
+
+def usage
+  puts <<~USAGE
+
+    Usage: bundle exec ruby database_samples.rb [command] [arguments]
+
+    Commands:
+      instance_config                           <instance_id> <instance_config_id> Get instance config
+      list_instance_configs                     List instance configs
+      list_databases                            <instance_id> List databases
+      create_database_with_default_leader       <instance_id> <database_id> <default_leader> Create database with default leader
+      update_database_with_default_leader       <instance_id> <database_id> <default_leader> Update database default leader
+      database_ddl                              <instance_id> <database_id> Get database DDL statements
+      query_information_schema_database_options <instance_id> <database_id> Get database information schema using option name
+
+    Environment variables:
+      GOOGLE_CLOUD_PROJECT must be set to your Google Cloud project ID
+  USAGE
+end
+
+def run_sample arguments
+  command = arguments.shift
+  project_id = ENV["GOOGLE_CLOUD_PROJECT"]
+
+  case command
```

Comment:
Can this be refactored because they have some common arguments?

Reply:
similar to: https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/spanner/spanner_samples.rb#L2175-L2225

Reply:
done using diffrent approach because first two samples has diffrent arguments parameters